test/test_functional_filtering.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import math
+import os
+import torch
+import torchaudio
+import torchaudio.functional as F
+import torchaudio.functional_sox_compatibility as FSOX
+import unittest
+import common_utils
+import time
+
+
+class TestFunctionalFiltering(unittest.TestCase):
+    test_dirpath, test_dir = common_utils.create_temp_assets_dir()
+
+    def test_lfilter_basic(self):
+        """
+        Create a very basic signal,
+        Then make a simple 4th order delay
+        The output should be same as the input but shifted
+        """
+
+        torch.random.manual_seed(42)
+        waveform = torch.rand(2, 44100 * 10)
+        b_coeffs = torch.tensor([0, 0, 0, 1], dtype=torch.float32)
+        a_coeffs = torch.tensor([1, 0, 0, 0], dtype=torch.float32)
+        output_waveform = F.lfilter(waveform, a_coeffs, b_coeffs)
+
+        assert torch.allclose(
+            waveform[:, 0:-3], output_waveform[:, 3:], atol=1e-5
+        )
+
+    def test_lfilter(self):
+        """
+        Design an IIR lowpass filter using scipy.signal filter design
+        https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.iirdesign.html#scipy.signal.iirdesign
+
+        from scipy.signal import iirdesign
```

Comment:
Each line of the example should start with a `>`, see [here](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L121) for instance.

Reply:
OK